lib/metadata/index.js

```diff
@@ -254,12 +284,88 @@ Metadata.prototype.getReplicas = function (keyspaceName, tokenBuffer) {
   if (!keyspaceName) {
     return [this.primaryReplicas[closestToken]];
   }
-  if (!keyspace.replicas) {
-    //Calculate replicas the first time for the keyspace
-    keyspace.replicas =
-      keyspace.tokenToReplica(this.tokenizer, this.ringTokensAsStrings, this.primaryReplicas, this.datacenters);
+
+  const replicas = this._getKeyspaceReplicas(keyspace);
+  return replicas[closestToken];
+};
+
+/**
+ * Gets the token ranges that define data distribution in the ring.
+ *
+ * @returns {Set<TokenRange>} The ranges of the ring.
```

Comment:
After using the API a bit I think it might be confusing that some methods return Array and others Set, should I change these to return Array?